futures/tests/async_await_macros.rs

```diff
@@ -308,6 +310,42 @@ fn select_on_mutable_borrowing_future_with_same_borrow_in_block_and_default() {
     });
 }
 
+#[test]
+#[allow(unused_assignments)]
+fn stream_select() {
+    // stream_select! macro
+    block_on(async {
+        let endless_ints = |i| stream::iter(vec![i].into_iter().cycle());
+
+        let mut endless_ones = stream_select!(endless_ints(1i32), stream::pending());
+        assert_eq!(endless_ones.next().await, Some(1));
+        assert_eq!(endless_ones.next().await, Some(1));
+
+        let mut finite_list = stream_select!(stream::iter(vec![1, 2, 3].into_iter()));
+        assert_eq!(finite_list.next().await, Some(1));
+        assert_eq!(finite_list.next().await, Some(2));
+        assert_eq!(finite_list.next().await, Some(3));
+        assert_eq!(finite_list.next().await, None);
```

Comment:
If stream_select takes one stream, it is more inefficient than simply calling next on that stream, right?
If so, it would probably be preferable to make that case an error.

Reply:
Reasonable, and corrected. This is ready for re-review!